python/lsst/pipe/tasks/make_direct_warp.py

```diff
@@ -408,15 +410,22 @@ def run(self, inputs, sky_info, visit_summary):
             )
             warpedExposure.setPsf(psfWarped)
 
+            # Calculate the valid polygon of the warped exposure.
+            polygon = Polygon(
+                [xyTransform.applyForward(Point2D(corner)) for corner in calexp.getBBox().getCorners()]
+            )
+            polygon = polygon.intersectionSingle(Box2D(warpedExposure.getBBox()))
+
             # Accumulate the partial warps in an online fashion.
             nGood = copyGoodPixels(
                 final_warp.maskedImage,
                 warpedExposure.maskedImage,
                 final_warp.mask.getPlaneBitMask(["NO_DATA"]),
             )
             ccdId = self.config.idGenerator.apply(dataId).catalog_id
-            inputRecorder.addCalExp(calexp, ccdId, nGood)
+            record = inputRecorder.addCalExp(calexp, ccdId, nGood)
             totalGoodPixels += nGood
+            record.setValidPolygon(polygon)
```

Comment:
It just seemed appropriate that a warp should be able to separate out the detectors. I do this calculation in the PSF-matching task if this information is not available, but I thought it'd be nice to persist this info with the warp itself.

Reply:
This actually seems to be causing harm. `CoaddPsf` is unable to find the average position with this being set.

Reply:
I'm not sure why that is, but it sounds like the safe thing to do is just remove it.

Reply:
Well, to be fair, it's causing issues in unit tests and it wasn't evident when I ran it on some DC2 data. The unit tests aren't robust, so it could actually be an issue with the tests. In any case, punting it to a different ticket since I don't _need_ this.